pkg/controller/servicediscovery/privatednsnamespace/hooks.go

```diff
@@ -0,0 +1,160 @@
+package privatednsnamespace
+
+import (
+	"context"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	cpresource "github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/servicediscovery/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+func (e *external) observeImpl(ctx context.Context, cpm cpresource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	ret := managed.ExternalObservation{}
+	cr, ok := cpm.(*svcapitypes.PrivateDNSNamespace)
```

Comment:
I think `mg` is the convention for `cpresource.Managed` variables (at least that's how it is named in other controllers. So I would suggest to stick to it.

Reply:
done